### PR TITLE
Add onKeyDown to TextField and TextArea

### PR DESCRIPTION
**Pull request by jennyscript**

### CODE REVIEW

**Comment by christianvuerings** [comment]
Deploy preview for *gestalt* ready!

Built with commit 23fb0a726b376691de25fe2ad27b7c363169fc0d

https://deploy-preview-303--gestalt.netlify.com

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :memo: CHANGELOG - <i>Please add a CHANGELOG.md entry for your changes</i>
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :file_folder: Bundlesize - <i>Bundle size increase detected, please review</i>
  </td>
    </tr>
  </tbody>
</table>


Filename | Size Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
gestalt.js | :small_red_triangle: 0.2% | :small_red_triangle: 0.0% | 261.37 KB | 261.98 KB | 58.72 KB | 58.75 KB
gestalt.es.js | :small_red_triangle: 0.2% | :small_red_triangle: 0.1% | 258.7 KB | 259.31 KB | 58.35 KB | 58.38 KB
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: docs/src/TextArea.doc.js [hunk 1]
```diff
@@ -44,16 +44,24 @@ card(
       },
       {
         name: 'onBlur',
-        type: '({ event: SyntheticFocusEvent<>, value: string }) => void',
+        type:
```
Comment (pfarejowicz) [inline docs/src/TextArea.doc.js]: These are going to cause Flow errors when bumping in pinboard - prepare yourselves, winter is coming

File: packages/gestalt/src/TextArea.js [hunk 2]
```diff
@@ -68,39 +82,46 @@ export default class TextArea extends React.Component<Props, State> {
...
-  handleChange = (event: SyntheticInputEvent<>) => {
-    if (event.target instanceof HTMLTextAreaElement) {
-      this.props.onChange({
-        event,
-        value: event.target.value,
-      });
+  handleChange = (event: SyntheticInputEvent<HTMLTextAreaElement>) => {
+    this.props.onChange({
+      event,
+      value: event.currentTarget.value,
```
Comment (pfarejowicz) [inline packages/gestalt/src/TextArea.js]: are we sure this is the same, `event.target.value` and `event.currentTarget.value`? i know there was once a `currentTarget` `target` change in pinboard and it caused a rollback/hotfix

**Pull request merged**